commit b8f4b96768c08e9273b67e720be0d6287f1ac291
Author: Ludovic Desroches <ludovic.desroches@microchip.com>
Date:   Mon Aug 20 10:54:45 2018 +0200

    mmc: android-goldfish: fix bad logic of sg_copy_{from,to}_buffer conversion
    
    [ Upstream commit 17e96d8516e31c3cb52cb8e2ee79d1d2e6948c11 ]
    
    The conversion to sg_copy_{from,to}_buffer has been done in the wrong
    way. sg_copy_to_buffer is a copy from an SG list to a linear buffer so
    it can't replace memcpy(dest, host->virt_base, data->sg->length) where
    dest is the virtual address of the SG. Same for sg_copy_from_buffer
    but in the opposite way.
    
    Signed-off-by: Ludovic Desroches <ludovic.desroches@microchip.com>
    Suggested-by: Douglas Gilbert <dgilbert@interlog.com>
    Fixes: 53d7e098ba08 ("mmc: android-goldfish: use sg_copy_{from,to}_buffer")
    Signed-off-by: Ulf Hansson <ulf.hansson@linaro.org>
    Signed-off-by: Sasha Levin <alexander.levin@microsoft.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

diff --git a/drivers/mmc/host/android-goldfish.c b/drivers/mmc/host/android-goldfish.c
index 294de177632c..61e4e2a213c9 100644
--- a/drivers/mmc/host/android-goldfish.c
+++ b/drivers/mmc/host/android-goldfish.c
@@ -217,7 +217,7 @@ static void goldfish_mmc_xfer_done(struct goldfish_mmc_host *host,
 			 * We don't really have DMA, so we need
 			 * to copy from our platform driver buffer
 			 */
-			sg_copy_to_buffer(data->sg, 1, host->virt_base,
+			sg_copy_from_buffer(data->sg, 1, host->virt_base,
 					data->sg->length);
 		}
 		host->data->bytes_xfered += data->sg->length;
@@ -393,7 +393,7 @@ static void goldfish_mmc_prepare_data(struct goldfish_mmc_host *host,
 		 * We don't really have DMA, so we need to copy to our
 		 * platform driver buffer
 		 */
-		sg_copy_from_buffer(data->sg, 1, host->virt_base,
+		sg_copy_to_buffer(data->sg, 1, host->virt_base,
 				data->sg->length);
 	}
 }
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1267_b8f4b96768c08e9273b67e720be0d6287f1ac291.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1267_b8f4b96768c08e9273b67e720be0d6287f1ac291.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
